resources/assets/components/CampaignSingle/index.js

```diff
@@ -24,15 +24,37 @@ class CampaignSingle extends React.Component {
     this.deletePost = this.deletePost.bind(this);
     this.filterPosts = this.filterPosts.bind(this);
     this.getPostsByPaginatedLink = this.getPostsByPaginatedLink.bind(this);
+    this.createTagSlug = this.createTagSlug.bind(this);
+    this.getPostsByTag = this.getPostsByTag.bind(this);
   }
 
   componentDidMount() {
     this.getPostsByStatus('accepted', this.props.campaign.id);
   }
 
-  // Filter posts based on status.
-  filterPosts(status) {
-    this.getPostsByStatus(status.toLowerCase(), this.props.campaign.id);
+  // Filter posts based on status or tag(s).
+  filterPosts(filter) {
+    // If the filter is a status, make API call to get posts by status.
+    if (filter === 'Pending' || filter === 'Accepted' || filter === 'Rejected') {
+      this.getPostsByStatus(filter.toLowerCase(), this.props.campaign.id);
+    } else {
```

Comment:
What happens if we want to filter by status AND tag. Could we potentially grab all the filter fields and then make one request with all of the filter values? If this feels out of scope for this PR, we could create a card to update the functionality. But we should be clear that you can only do one or the other.

Reply:
Ah I see that below, we just do one large select list with filter options so we only allow one filter. In this case it is fine, but I definitely think we should update this functionality so that the `status` filter and `tags` filter are separate. We should be able to do something like give me all `Accepted` posts that have a `good-photo` tag. So can you make a note to make a new card for this work and we can prioritize that later?

Reply:
Yes - I'll make a card for this now! cc @ngjo